contracts/middleware/BaseMiddleware.sol

```diff
@@ -3,28 +3,37 @@ pragma solidity ^0.8.24;
 
 import {Hooks} from "@uniswap/v4-core/src/libraries/Hooks.sol";
 import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
-import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
-import {BalanceDelta} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
-import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
-import {BeforeSwapDelta} from "@uniswap/v4-core/src/types/BeforeSwapDelta.sol";
 import {Proxy} from "@openzeppelin/contracts/proxy/Proxy.sol";
 
-contract BaseMiddleware is Proxy {
+/**
+ * @notice Proxy contract that forwards calls to an implementation contract.
+ * @dev Inheret this contract and implement hook functions to create a middleware.
+ */
+abstract contract BaseMiddleware is Proxy {
     /// @notice The address of the pool manager
-    IPoolManager public immutable poolManager;
+    /// @dev Use in middleware implementations to access the pool manager
+    IPoolManager public immutable manager;
+
+    /// @notice The address of the implementation contract. All calls to this contract will be forwarded to implementation.
     address public immutable implementation;
 
-    constructor(IPoolManager _poolManager, address _impl) {
-        poolManager = _poolManager;
+    error FlagsMismatch();
+
+    constructor(IPoolManager _manager, address _impl) {
+        _ensureValidFlags(_impl);
+        manager = _manager;
         implementation = _impl;
     }
 
     function _implementation() internal view override returns (address) {
         return implementation;
     }
 
-    // yo i wanna delete this function but how do i remove this warning
-    receive() external payable {
-        _delegate(_implementation());
+    /// @notice Ensure that the implementation contract has the correct hook flags.
+    /// @dev Override to enforce hook flags.
+    function _ensureValidFlags(address _impl) internal view virtual {
```

Comment:
Is there a reason the implementation needs to have the same flags as the middleware?

Reply:
no, will remove this actually